### PR TITLE
Issue 10 - Ananke should allow non integers for metric values

### PR DESCRIPTION
**Pull request by joepusateri**
Changed the type of metric values from Long to Double to allow non-integer values.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 group = 'com.netuitive'
-version = '0.1.24'
+version = '0.1.25'
```
Comment (TheConnMan) [inline build.gradle]: Preferably the version is only updated directly on the `develop` branch right before merging into `master`. This allows for multiple PRs to be compiled under a single release.

File: src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java [hunk 2]
```diff
@@ -27,10 +27,15 @@ public DecrementRequest withTags(List<Tag> tags) {
         return this.tags == tags ? this : new DecrementRequest(this.metric, this.value, tags, this.sampleRate);
     }
 
-    public DecrementRequest withValue(Long value) {
+    public DecrementRequest withValue(Double value) {
```
Comment (jking13) [inline src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java]: we should override these methods with Long and then do a value.doubleValue() on it internally so we don't make any breaking changes.

File: src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java [hunk 3]
```diff
@@ -27,10 +27,15 @@ public DecrementRequest withTags(List<Tag> tags) {
...
         return this.value == value ? this : new DecrementRequest(this.metric, value, this.tags, this.sampleRate);
     }
 
+    public DecrementRequest withValue(Long value) {
+    	Double dvalue = (value == null ? null : value.doubleValue());
```
Comment (TheConnMan) [inline src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java]: Missing a space here.

File: src/main/java/com/netuitive/ananke/statsd/client/request/GaugeRequest.java [hunk 4]
```diff
@@ -27,6 +27,11 @@ public GaugeRequest withTags(List<Tag> tags) {
     }
 
     public GaugeRequest withValue(Long value) {
+    	Double dvalue = (value == null ? null : value.doubleValue());
```
Comment (TheConnMan) [inline src/main/java/com/netuitive/ananke/statsd/client/request/GaugeRequest.java]: Missing a space here.

File: src/main/java/com/netuitive/ananke/statsd/client/request/HistogramRequest.java [hunk 5]
```diff
@@ -28,6 +28,11 @@ public HistogramRequest withTags(List<Tag> tags) {
     }
 
     public HistogramRequest withValue(Long value) {
+    	Double dvalue = (value == null ? null : value.doubleValue());
```
Comment (TheConnMan) [inline src/main/java/com/netuitive/ananke/statsd/client/request/HistogramRequest.java]: Missing a space here.

File: src/main/java/com/netuitive/ananke/statsd/client/request/SetRequest.java [hunk 6]
```diff
@@ -28,6 +28,11 @@ public SetRequest withTags(List<Tag> tags) {
     }
 
     public SetRequest withValue(Long value) {
+    	Double dvalue = (value == null ? null : value.doubleValue());
```
Comment (TheConnMan) [inline src/main/java/com/netuitive/ananke/statsd/client/request/SetRequest.java]: Space.

File: src/main/java/com/netuitive/ananke/statsd/client/request/IncrementRequest.java [hunk 7]
```diff
@@ -28,6 +28,11 @@ public IncrementRequest withTags(List<Tag> tags) {
     }
 
     public IncrementRequest withValue(Long value) {
+    	Double dvalue = (value == null ? null : value.doubleValue());
```
Comment (TheConnMan) [inline src/main/java/com/netuitive/ananke/statsd/client/request/IncrementRequest.java]: Space.

File: src/main/java/com/netuitive/ananke/statsd/client/request/TimingRequest.java [hunk 8]
```diff
@@ -27,6 +27,11 @@ public TimingRequest withTags(List<Tag> tags) {
     }
 
     public TimingRequest withValue(Long value) {
+    	Double dvalue = (value == null ? null : value.doubleValue());
```
Comment (TheConnMan) [inline src/main/java/com/netuitive/ananke/statsd/client/request/TimingRequest.java]: Space.

File: src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java (hunk 2 above)
Comment (joepusateri) [inline src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java]: That was already done.  Line 30 has the old Long signature and Line 35 has the new Double signature for withValue().

Comment (jking13) [inline src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java]: Ahh sorry I missed it.

File: build.gradle (hunk 1 above)
Comment (joepusateri) [inline build.gradle]: Got it. I'll reverse this.

File: src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java [hunk 9]
```diff
@@ -12,7 +12,7 @@
     public DecrementRequest() {
     }
 
-    public DecrementRequest(String metric, Long value, List<Tag> tags, Long sampleRate) {
+    public DecrementRequest(String metric, Double value, List<Tag> tags, Long sampleRate) {
```
Comment (TheConnMan) [inline src/main/java/com/netuitive/ananke/statsd/client/request/DecrementRequest.java]: What about backwards compatibility for this too? I would expect there to be an overloaded constructor such as:
```
public DecrementRequest(String metric, Long value, List<Tag> tags, Long sampleRate) {
    this(metric, value == null ? null : value.doubleValue(), tags, sampleRate);
}
```

**Pull request merged**